src/test/java/com/faforever/client/mod/ModServiceTest.java

```diff
@@ -405,6 +408,61 @@ public void testLoadThumbnail() throws MalformedURLException {
     verify(assetService).loadAndCacheImage(eq(modVersion.getThumbnailUrl()), eq(Paths.get("mods")), any());
   }
 
+  @Test
+  public void testUpdateModsWithUpdatedMod() throws IOException, ExecutionException, InterruptedException {
+    ModVersion modVersion = ModVersionBuilder.create().defaultValues().get();
+    Mod mod = new Mod();
+    modVersion.setMod(mod);
+    mod.setLatestVersion(modVersion);
+
+    when(fafService.getModVersion(eq(modVersion.getUid())))
+        .thenReturn(CompletableFuture.completedFuture(modVersion));
+
+    Files.createFile(gamePrefsPath);
+
+
+    List<ModVersion> modVersions = instance.updateModVersionsAndActivateThose(List.of(modVersion)).get();
+
+    verify(taskService, times(1)).submitTask(any());
```

Comment:
add verification that the modService never tries to download the mod in this case

Reply:
This is the verifyication

Reply:
I can make it more explicit

Reply:
Ah yeah I must have missed that as I was expecting something on modService